__tests__/api.test.js

```diff
@@ -292,13 +314,57 @@ test('ignore extra properties with a status 201', () => {
     .send(newComment)
     .expect(201)
     .then(res => {
-        expect(res.body.comment).toMatchObject({
-            comment_id: expect.any(Number),
-            body: 'This is a new comment',  
-            article_id: 1,
-            author: 'butter_bridge',
-            votes: expect.any(Number),
-            created_at: expect.any(String)
+            expect(res.body.comment).toMatchObject({
+                comment_id: expect.any(Number),
+                body: 'This is a new comment',  
+                article_id: 1,
+                author: 'butter_bridge',
+                votes: expect.any(Number),
+                created_at: expect.any(String)
+            })
+        })
+    })
+})
+
+describe('PATCH error handling', () => {
+    test('responds with a 400 when passed a invalid data type', () => {
+        return request(app)
+        .patch('/api/articles/string')
+        .send({ inc_votes: 1 })
+        .expect(400)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Bad request');
+        })
+    })
+    test('responds with a 404 status code when given an invalid ID', () => {
+        return request(app)
+        .patch('/api/articles/99999999')
+        .send({ inc_votes: 1 })
+        .expect(404)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Article not found');
+        })
+    })
+    test('responds with a 404 status code when given a non existent article id', () => {
+        return request(app)
+        .patch('/api/articles/999999')
+        .send({ inc_votes: 1 })
+        .expect(404)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Article not found');
+        })
+    })
+    test('ignores irrelevant properties in the request body', () => { 
+        return request(app)
+        .patch('/api/articles/1')
+        .send({ inc_votes: 1, comment_id: 1 })
+        .expect(200)
+        .then(res => {
+            expect(res.body.article).toMatchObject({
+                votes: 101
         })
```

Comment:
Just needs one more test for if inc_votes is missing - either respond with a bad request error, or make no changes and send a 200, both valid